Yosemite/Yosemite/PointOfSale/PointOfSaleItemServiceProtocol.swift

```diff
@@ -1,36 +1,48 @@
-public protocol POSItem {
-    var itemID: UUID { get }
-    var productID: Int64 { get }
+public protocol PointOfSaleItemDisplayable: Identifiable {
+    var id: UUID { get }
     var name: String { get }
-    var price: String { get }
     var formattedPrice: String { get }
-    var itemCategories: [String] { get }
     var productImageSource: String? { get }
-    var productType: ProductType { get }
+
+    func isEqual(to other: any PointOfSaleItemDisplayable) -> Bool
 }
 
-extension POSItem {
-    // Equatable conformance
-    static func == (lhs: Self, rhs: Self) -> Bool {
-        lhs.itemID == rhs.itemID &&
-        lhs.productID == rhs.productID &&
-        lhs.name == rhs.name &&
-        lhs.price == rhs.price &&
-        lhs.formattedPrice == rhs.formattedPrice &&
-        lhs.itemCategories == rhs.itemCategories &&
-        lhs.productImageSource == rhs.productImageSource &&
-        lhs.productType == rhs.productType
+public extension PointOfSaleItemDisplayable where Self: Equatable {
+    func isEqual(to other: any PointOfSaleItemDisplayable) -> Bool {
+        guard let other = other as? Self else { return false }
+        return self == other
     }
 }
 
+public extension Sequence where Element == any PointOfSaleItemDisplayable {
+    func isEqual(to other: any Sequence<Element>) -> Bool {
+        let lhsArray = Array(self)
+        let rhsArray = Array(other)
```

Comment:
This might be conflating sequence and Array – it's an order-specific comparison, which is what we want but perhaps it shouldn't be an extension of every Sequence of Displayables... not sure.